src/auth/views/auth/join-almost/join-almost.ts

```diff
@@ -8,21 +8,85 @@ import { AppProgressPoller } from '../../../../lib/gj-lib-client/components/prog
 import { Growls } from '../../../../lib/gj-lib-client/components/growls/growls.service';
 import { Store } from '../../../store/index';
 import { BaseRouteComponent } from '../../../../lib/gj-lib-client/components/route/route-component';
+import { AppLoading } from '../../../../lib/gj-lib-client/vue/components/loading/loading';
+import { AppJolticon } from '../../../../lib/gj-lib-client/vue/components/jolticon/jolticon';
 
 @View
 @Component({
 	name: 'RouteJoinAlmost',
 	components: {
 		AppProgressPoller,
+		AppLoading,
+		AppJolticon,
 	},
 })
 export default class RouteJoinAlmost extends BaseRouteComponent {
 	@State credentials: Store['credentials'];
 
+	doingCaptcha = true;
+	recaptchaReady = false;
+	recaptchaInterval: NodeJS.Timer = null as any;
+
+	serverErrors: {
+		token: boolean;
+		captcha: boolean;
+		email: boolean;
+		username: boolean;
+		'rate-limit': boolean;
+	} = null as any;
+
+	$refs: {
+		recaptcha: HTMLDivElement;
+	};
+
 	get routeTitle() {
 		return this.$gettext('auth.join.almost.page_title');
 	}
 
+	routeInit() {
+		this.doingCaptcha = !!this.$route.params.token;
+
+		this.recaptchaInterval = setInterval(() => {
+			if (grecaptcha) {
```

Comment:
Instead of recaptcha interval mess - make a component that is in charge of loading. Look at facebook/twitter/youtube sdks for reference.